src/network/networkd-sysctl.c

```diff
@@ -315,11 +389,40 @@ static int link_set_ipv4_promote_secondaries(Link *link) {
         return sysctl_write_ip_property_boolean(AF_INET, link->ifname, "promote_secondaries", true, &link->sysctls);
 }
 
+static void link_watch_sysctl(Link *link) {
+        sd_event_source **source;
+
+        assert(link);
+
+        source = link->sysctl_event_sources;
+
+        FOREACH_STRING(p,
+                "/proc/sys/net/ipv4/conf",
+                "/proc/sys/net/ipv6/conf") {
+                _cleanup_free_ char *path = path_join(p, link->ifname);
+                int r;
+
+                if (!path)
+                        return;
+
+                /* We're reconfiguring the link, an inotify watcher is already present for this path */
+                if (*source)
+                        continue;
+
+                r = sd_event_add_inotify(link->manager->event, source, path, IN_CLOSE_WRITE | IN_ONLYDIR, sysctl_inotify_link, link);
+                if (r < 0)
+                        log_link_warning_errno(link, errno, "Failed to watch sysctl %s: %m", path);
```

Comment:
2nd argument should be `r`, not `errno`.

Like with the discussion above, logging at warning level is fine, but please suffix with "ignoring"; i.e. `… to watch sysctl '%s', ignoring: %m"`